### PR TITLE
fix(frontend-pr-analysis): detect vitest version for coverage reporter flag

### LINKED ISSUE #215: fix: frontend-pr-analysis passes --coverageReporters CLI flag rejected by Vitest 3
## Affected Workflow

TypeScript (ci, release, frontend-pr-analysis)

## Workflow Version / Ref

`@v1.18.1`

## Bug Description

The `frontend-pr-analysis` workflow passes `--coverageReporters` as a CLI flag when running Vitest. Vitest 3 removed support for this CLI option, causing the test step to fail with `Unknown option --coverageReporters`.

This blocks any repository that upgrades to Vitest 3 from passing CI, since the shared workflow injects an unsupported flag.

## Steps to Reproduce

1. Repository using `frontend-pr-analysis.yml@v1.18.1`
2. Upgrade Vitest to v3.x in the consuming repository
3. Open a PR to trigger the workflow
4. Test step fails with `Unknown option --coverageReporters`

Reference PR: https://github.com/LerianStudio/lerian-map/pull/144

## Expected Behavior

The workflow should either:
- Configure coverage reporters via `vitest.config.ts` (`coverage.reporter`) instead of CLI flags, or
- Detect the Vitest major version and omit the flag for v3+, or
- Accept an input parameter to disable/customize coverage reporter flags

## Relevant Logs / Error Output

```
Unknown option --coverageReporters
```

## Caller Workflow Configuration

```yaml
uses: LerianStudio/github-actions-shared-workflows/.github/workflows/frontend-pr-analysis.yml@v1.18.1
```

## Runner OS

ubuntu-latest

## Checklist

- [x] I searched existing issues and this is not a duplicate.
- [x] I am using a supported version/ref of the workflow.
- [x] I have included relevant logs and configuration above.

## Additional Context

Vitest 3 changelog confirms `--coverageReporters` was removed as a CLI option. The recommended approach is to set `coverage.reporter` in the Vitest config file instead.

### PR DESCRIPTION
**Pull request by bedatty**
<table border="0" cellspacing="0" cellpadding="0">
  <tr>
    <td>Lerian</td>
    <td><h1>GitHub Actions Shared Workflows</h1></td>
  </tr>
</table>

---

## Description

Vitest 3 removed the `--coverageReporters` CLI flag, causing the `frontend-pr-analysis` workflow to fail with `Unknown option --coverageReporters` for any repository that upgrades to Vitest 3+.

This PR detects the installed Vitest major version at runtime and selects the appropriate CLI syntax:
- **Vitest 3+**: `--coverage.reporter=X` (dot notation)
- **Vitest 2 / Jest / other**: `--coverageReporters=X` (original syntax)

If Vitest is not installed (e.g., project uses Jest), the detection safely falls back to the original flags.

## Type of Change

- [x] `fix`: Bug fix in a workflow (incorrect behavior, broken step, wrong condition)

## Breaking Changes

None.

## Testing

- [x] YAML syntax validated locally
- [ ] Triggered a real workflow run on a caller repository using `@develop` or the beta tag
- [x] Verified all existing inputs still work with default values
- [x] Confirmed no secrets or tokens are printed in logs
- [x] Checked that unrelated workflows are not affected

**Caller repo / workflow run:** https://github.com/LerianStudio/lerian-map/pull/144

## Related Issues

Closes #215

## Summary by CodeRabbit

* **Chores**
  * CI workflows pinned to fixed action versions for more reproducible runs.
  * Safer handling and quoting of CI variables and outputs; improved PR diff and file-change detection logic.
  * Consolidated CI output writes into grouped blocks for more reliable logging.

* **Tests**
  * Test workflow detects Vitest major version and builds appropriate coverage flags dynamically.
  * Coverage reporting and summary generation now use injected env vars and unified output handling.

### CODE REVIEW

**Comment by lerian-studio** [comment]
## 🔍 Lint Analysis

| Check | Files Scanned | Status |
|-------|:-------------:|:------:|
| YAML Lint | 1 file(s) | ✅ success |
| Action Lint | 1 file(s) | ✅ success |
| Pinned Actions | 1 file(s) | ✅ success |
| Markdown Link Check | _no changes_ | ⏭️ skipped |
| Spelling Check | 1 file(s) | ✅ success |
| Shell Check | 1 file(s) | ✅ success |
| README Check | 1 file(s) | ✅ success |
| Composite Schema | _no changes_ | ⏭️ skipped |
---
<sub>🔍 [View full scan logs](https://github.com/LerianStudio/github-actions-shared-workflows/actions/runs/24417083575)</sub>

**Comment by lerian-studio** [comment]
## 🛡️ CodeQL Analysis Results

**Languages analyzed:** `actions`

✅ No security issues found.

---

<sub>🔍 [View full scan logs](https://github.com/LerianStudio/github-actions-shared-workflows/actions/runs/24417083575) | 🛡️ [Security tab](https://github.com/LerianStudio/github-actions-shared-workflows/security/code-scanning?query=pr%3A219+is%3Aopen)</sub>

File: .github/workflows/frontend-pr-analysis.yml [hunk 1]
```diff
@@ -364,10 +364,20 @@ jobs:
...
+            COVERAGE_FLAGS="--coverageReporters=text --coverageReporters=json-summary --coverageReporters=lcov"
+          fi
+
           case "${{ inputs.package_manager }}" in
-            yarn) yarn test --coverage --coverageReporters=text --coverageReporters=json-summary --coverageReporters=lcov ;;
-            pnpm) pnpm test --coverage --coverageReporters=text --coverageReporters=json-summary --coverageReporters=lcov ;;
-            *) npm test -- --coverage --coverageReporters=text --coverageReporters=json-summary --coverageReporters=lcov ;;
+            yarn) yarn test --coverage $COVERAGE_FLAGS ;;
+            pnpm) pnpm test --coverage $COVERAGE_FLAGS ;;
+            *) npm test -- --coverage $COVERAGE_FLAGS ;;
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Jest **does** support `--coverageReporters` as a CLI flag — it's documented at [jestjs.io/docs/cli](https://jestjs.io/docs/cli). The web search result cited here is inaccurate.

Similarly, Vitest 2 supported `--coverageReporters` as a Jest-compatible flag — the issue reporter (#215) confirms it worked on v2 and broke on v3.

The current logic handles all real-world cases correctly:
- **Vitest not installed** (VITEST_MAJOR=0) → `--coverageReporters` → valid for Jest
- **Vitest 2** → `--coverageReporters` → valid for Vitest 2
- **Vitest 3+** → `--coverage.reporter` → valid for Vitest 3+

The edge case (Vitest 3 in node_modules but test script runs Jest) is impractical in real monorepos where `working_dir` targets a specific app. Adding a `TEST_SCRIPT == *vitest*` heuristic would also fail for wrapper scripts or indirect invocations.

File: .github/workflows/frontend-pr-analysis.yml [hunk 2]
```diff
@@ -112,12 +112,11 @@ jobs:
           if [[ "${{ github.event_name }}" == "pull_request" ]]; then
             BASE_SHA="${{ github.event.pull_request.base.sha }}"
             HEAD_SHA="${{ github.event.pull_request.head.sha }}"
-            git fetch origin $BASE_SHA --depth=1 2>/dev/null || true
-            FILES=$(git diff --name-only $BASE_SHA $HEAD_SHA 2>/dev/null || git diff --name-only origin/${{ github.base_ref }}...HEAD)
+            git fetch origin "$BASE_SHA" --depth=1 2>/dev/null || true
+            FILES=$(git diff --name-only "$BASE_SHA" "$HEAD_SHA" 2>/dev/null || git diff --name-only origin/${{ github.base_ref }}...HEAD)
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Already addressed — all `${{ }}` expressions in this step were mapped to `env:` vars in commit 21a8667.

File: .github/workflows/frontend-pr-analysis.yml [hunk 3]
```diff
@@ -409,23 +418,23 @@ jobs:
...
             echo "Total line coverage: $COVERAGE%"
           else
-            echo "coverage=0" >> $GITHUB_OUTPUT
-            echo "statements=0" >> $GITHUB_OUTPUT
-            echo "branches=0" >> $GITHUB_OUTPUT
-            echo "functions=0" >> $GITHUB_OUTPUT
+            echo "coverage=0" >> "$GITHUB_OUTPUT"
+            echo "statements=0" >> "$GITHUB_OUTPUT"
+            echo "branches=0" >> "$GITHUB_OUTPUT"
+            echo "functions=0" >> "$GITHUB_OUTPUT"
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Already addressed — grouped with `{ } >> "$GITHUB_OUTPUT"` in commit 7abd646.

File: .github/workflows/frontend-pr-analysis.yml [hunk 4]
```diff
@@ -112,12 +112,11 @@
           if [[ "${{ github.event_name }}" == "pull_request" ]]; then
             BASE_SHA="${{ github.event.pull_request.base.sha }}"
             HEAD_SHA="${{ github.event.pull_request.head.sha }}"
-            git fetch origin $BASE_SHA --depth=1 2>/dev/null || true
-            FILES=$(git diff --name-only $BASE_SHA $HEAD_SHA 2>/dev/null || git diff --name-only origin/${{ github.base_ref }}...HEAD)
+            git fetch origin "$BASE_SHA" --depth=1 2>/dev/null || true
+            FILES=$(git diff --name-only "$BASE_SHA" "$HEAD_SHA" 2>/dev/null || git diff --name-only origin/${{ github.base_ref }}...HEAD)
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Addressed — all `${{ }}` expressions in this step mapped to `env:` vars in commit 21a8667.

File: .github/workflows/frontend-pr-analysis.yml [hunk 5]
```diff
@@ -488,15 +501,17 @@
...
-          echo "" >> $GITHUB_STEP_SUMMARY
-          echo "| Metric | Value |" >> $GITHUB_STEP_SUMMARY
-          echo "|--------|-------|" >> $GITHUB_STEP_SUMMARY
-          echo "| Lines | ${{ steps.coverage.outputs.coverage }}% |" >> $GITHUB_STEP_SUMMARY
-          echo "| Statements | ${{ steps.coverage.outputs.statements }}% |" >> $GITHUB_STEP_SUMMARY
-          echo "| Branches | ${{ steps.coverage.outputs.branches }}% |" >> $GITHUB_STEP_SUMMARY
-          echo "| Functions | ${{ steps.coverage.outputs.functions }}% |" >> $GITHUB_STEP_SUMMARY
-          echo "| Threshold | ${{ inputs.coverage_threshold }}% |" >> $GITHUB_STEP_SUMMARY
+          {
+            echo "## Coverage Summary: ${{ matrix.app.name }}"
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Addressed — `${{ matrix.app.name }}` mapped to `env.APP_NAME` in commit 21a8667.

File: .github/workflows/frontend-pr-analysis.yml [hunk 6]
```diff
@@ -488,15 +501,17 @@
...
+            echo ""
+            echo "| Metric | Value |"
+            echo "|--------|-------|"
+            echo "| Lines | ${{ steps.coverage.outputs.coverage }}% |"
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Addressed — mapped to `env.COV_LINES` in commit 21a8667.

File: .github/workflows/frontend-pr-analysis.yml [hunk 7]
```diff
@@ -488,15 +501,17 @@
...
+            echo "| Statements | ${{ steps.coverage.outputs.statements }}% |"
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Addressed — mapped to `env.COV_STATEMENTS` in commit 21a8667.

File: .github/workflows/frontend-pr-analysis.yml [hunk 8]
```diff
@@ -488,15 +501,17 @@
...
+            echo "| Branches | ${{ steps.coverage.outputs.branches }}% |"
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Addressed — mapped to `env.COV_BRANCHES` in commit 21a8667.

File: .github/workflows/frontend-pr-analysis.yml [hunk 9]
```diff
@@ -488,15 +501,17 @@
...
+            echo "| Functions | ${{ steps.coverage.outputs.functions }}% |"
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Addressed — mapped to `env.COV_FUNCTIONS` in commit 21a8667.

File: .github/workflows/frontend-pr-analysis.yml [hunk 10]
```diff
@@ -488,15 +501,17 @@
...
+            echo "| Threshold | ${{ inputs.coverage_threshold }}% |"
```
[the remark this answers is not in the document]
Comment (bedatty) [reply .github/workflows/frontend-pr-analysis.yml]: Addressed — mapped to `env.COV_THRESHOLD` in commit 21a8667.
